bindings/python/pymongoarrow/api.py

```diff
@@ -264,8 +270,15 @@ def _tabular_generator(tabular):
             for row in i.to_pylist():
                 yield row
     elif isinstance(tabular, DataFrame):
-        for i in tabular.to_dict("records"):
-            yield i
+        for row in tabular.to_dict("records"):
+            yield row
+    elif isinstance(tabular, dict):
+        iter_dict = {k: np.nditer(v) for k, v in tabular.items()}
```

Comment:
Can we not iterate over this directly rather than explicitly calling next?

Reply:
That seems unintuitive to me. Would that not just be calling next, but in a different and less clear way? We are going to have to call  ``ndarray.item()`` anyways, so I don't think that there is any way to avoid the dictionary comprehension. I also cannot figure out how to implement it your way. With a for loop it calls next on the container, but not on the individual elements thereof. So we basically are forced to call next ourselves, if I understand it correctly.

Reply:
Fair enough